src/editline.c

```diff
@@ -1034,6 +1034,30 @@ static el_status_t meta(void)
         return CSeof;
 
 #ifdef CONFIG_ANSI_ARROWS
+    /* See: https://en.wikipedia.org/wiki/ANSI_escape_code */
+    /* Recognize ANSI escapes for `Meta+Left` and `Meta+Right`. */
+    if (c == '\e') {
```

Comment:
FWIW, I tried another Linux system with Konsole and it has several builtin settings:

- Its default: "XFree 4" bindings, sends Meta+Left=`\e[1;3D`, Meta+Right=`\e[1;3C`
- Linux console: Meta+Left=`\e[D`, Meta+Right=`\e[C` (I think equivalent to just Left/Right)
- Solaris console: Meta+Left=`\eOD`, Meta+Right=`\eOC`
- macOS: Meta+Left=`\e[1;3D`, Meta+Right=`\e[1;3C`

Meanwhile on actual macOS, I get Meta+Left=`\eb`, Meta+Right=`\ef`...

I'm not sure if editline needs/wants to support _all_ of those different cases, but at least the last one is already covered by `MetaMap` bindings. I am not sure what OS / terminal editor send `\e\e[C` / `\e\e[D`, or if there are also other "reasonable defaults", but is it worth trying to support as many of these as reasonable?

(btw, they were more consistent about Ctrl-Left/Ctrl-Right, with the exception of Solaris/Linux console which each sent the same sequence as their Meta bindings).

Reply:
Ohh, actually I just realized something! The `\e\e[C` keys are the same as `\e`+Left / `\e`+Right... I think a `\e` prefix is normally meant to treated the same as an actual meta char (i.e. [`0x80 | c`](https://github.com/troglobit/editline/blob/ac81ca6d4b15b5da64c6f8bfb6026fdb1c7c3687/include/editline.h#L30-L32)).

Maybe the check `if (rl_meta_chars && ISMETA(c))` to call `meta()`, should be updated to something like this? No idea if that would break other uses of `\e`?
```c
    if (rl_meta_chars) {
		if (ISMETA(c)) {
        	tty_push(UNMETA(c));
			return meta();
		}
		if (c == '\e') {
			return meta();
		}
    }
```

Reply:
- [ ] Konsole (XFree 4): Meta+Left=`\e[1;3D`, Meta+Right=`\e[1;3C`
- [ ] Konsole (Linux): Meta+Left=`\e[D`, Meta+Right=`\e[C`
- [ ] Konsole (Solaris): Meta+Left=`\eOD`, Meta+Right=`\eOC`
- [ ] macOS: Meta+Left=`\e[1;3D`, Meta+Right=`\e[1;3C`
- [ ] macOS (iTerm2): Meta+Left=\e\e[D, Meta+Right=\e\e[C
- [ ] macOS: Meta+Left=`\eb`, Meta+Right=`\ef`
- [ ] Meta+Left=`\e[1;3D`, Meta+Right=`\e[1;3C`

Reply:
After reading https://en.wikipedia.org/wiki/ANSI_escape_code#Terminal_input_sequences more carefully, I think `\e[1;9D` / `\e[1;9C` might also possible if the sequence is sent as Meta intead of Alt (see "the modifier is the sum of the following numbers" table).

It doesn't exactly explain  the `1;` component though, but I _think_ that just means "do it 1 time" based on some other resources I've seen, e.g. `\e[1;D` (== `\e[D`) is "one char left" , `\e[1;3D` is "one column left". etc. In practice these probably are always either `1` or omitted, I would guess.

Reply:
Okay, I went through the code and checked that all of these sequences are supported, and pushed 4c4455353a0a88bee09d5f27c28f81f747682fed to add support for `\e[1;3C` and `\e[1;3D`. A couple of these are interpreted as non-meta arrow keys (like `\e[D` and `\e[C`) so I didn't mess with those.